### DISCUSSION TITLE
USB serial not showing on LG C2

### DISCUSSION THREAD

**Discussion by jrhrsmit** [awawa-dev/HyperSerialESP32#35] [Q&A]
Not sure if this is the right place to ask this, but it appears that my LG C2 can't load the driver for the USB to serial converter on the ESP32.
After connecting, the device is detected, but not linked to any serial port (like `/dev/ttyUSB0`).

```bash
root@LGwebOSTV:~# dmesg | grep -i usb
[ 8370.608208] usb 3-1: USB disconnect, device number 6
[ 8373.111113] usb 3-1: new full-speed USB device number 7 using xhci-hcd
[ 8373.285690] usb 3-1: New USB device found, idVendor=1a86, idProduct=7523, bcdDevice=81.34
[ 8373.285694] usb 3-1: New USB device strings: Mfr=0, Product=2, SerialNumber=0
[ 8373.285696] usb 3-1: Product: USB Serial
```
On my laptop it does appear to work:
```bash
[24818.042767] usb 2-7: new full-speed USB device number 21 using xhci_hcd
[24818.184695] usb 2-7: New USB device found, idVendor=1a86, idProduct=7523, bcdDevice=81.34
[24818.184706] usb 2-7: New USB device strings: Mfr=0, Product=2, SerialNumber=0
[24818.184710] usb 2-7: Product: USB Serial
[24818.190786] usb 2-7: ch341-uart converter now attached to ttyUSB0
```
So the main difference is that last line, where it attaches it to ttyUSB0.
Afterwards I can read the serial port at 2Mbaud:
```bash
$ picocom /dev/ttyUSB0 -b 2000000
picocom v3.1

port is        : /dev/ttyUSB0
flowcontrol    : none
baudrate is    : 2000000
(...)

Type [C-a] [C-h] to see available commands
Terminal ready

Welcome!
Awa driver 9.
SPI APA102 compatible type (BGR).
```

I have a similar issue with an ESP8266.
The raspberry pi pico does seem to get attached on the LG C2 (but it shows as `/dev/ttyACM0` instead of `/dev/ttyUSB0`, so maybe different driver?), but that platform is not supported :)

Is there some trick to getting the `/dev/ttyUSB0` to appear? Do I need to modprobe some driver?

**Comment by awawa-dev** [comment]
Hi
You are probably using an board equipped with a ch340. I know there are attempts to inject a supporting driver into the webos kernel (no native drivers for ch340 in webos), but so far I haven't heard a success story that worked with the LED driver, so at least I can't help with that (dont use webos root anymore).

When it comes to webos, I know from reports that two boards work with most versions of webos: ESP32 MH-ET Live (CH9102x version) and ESP32-S2 lolin mini (CDC). Due to the fact that webos tends to periodically (e.g. when using the youtube application, probably it searches for user devices like keyboard, mouse or USB harddrives) reset the usb bus, a physical one will probably be better than a virtual CDC. Don't get attached to webos paths: the same device may get a different path after reboot. Therefore, in this case it is recommended to set the path to 'auto' and enable 'esp32/esp8266 handshake' so that HyperHDR can find the right device by itself. It may also be helpful to enable auto-resume in case the esp device is not ready the first time, eg ESP32-S2 lolin with CDC takes much longer to initialize than physical CH9102x.

**Comment by jrhrsmit** [comment]
Thank you so much for this info, I'll order an ESP with the CH9102x.
